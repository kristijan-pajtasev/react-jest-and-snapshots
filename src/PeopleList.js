import React from 'react';
import logo from './logo.svg';
import './PeopleList.css';
import Person from "./Person.";

const PeopleList = (props) => {
  return (
      <div>
        <div>People:</div>
        <ul>
          { (props.people || []).map(person => <Person handleClick={props.handleClick} {...person} />) }
        </ul>
      </div>
  );
}

export default PeopleList;
