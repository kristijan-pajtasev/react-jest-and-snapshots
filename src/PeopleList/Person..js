import React from 'react';

const Person = (props) => (
    <div data-testid={`person-${props.id}`} onClick={props.handleClick}>
        <div>{props.firstName}</div>
        <div>{props.secondName}</div>
    </div>
);

export default Person;
