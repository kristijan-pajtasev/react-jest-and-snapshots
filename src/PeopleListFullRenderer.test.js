import React from 'react';
// import ReactDOM from 'react-dom';
import PeopleList from './PeopleList';
import TestRenderer from 'react-test-renderer';


const pplList = [
    {firstName: "John", lastName: "Doe"}
];

describe("People list with full renderer ", () => {

    it('renders without crashing', () => {

        const tree = TestRenderer
            .create(<PeopleList people={pplList}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
