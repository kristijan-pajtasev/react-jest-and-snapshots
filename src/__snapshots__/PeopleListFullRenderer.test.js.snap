// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People list with full renderer  renders without crashing 1`] = `
<div>
  <div>
    People:
  </div>
  <ul>
    <div>
      <div>
        John
      </div>
      <div />
    </div>
  </ul>
</div>
`;
